### ISSUE TITLE
Training from scratch results in gibberish

### ISSUE THREAD

**Issue by Respaired** [lucasnewman/nanospeech#11]
Hi again. I'm trying to train Nanospeech using 500-800 hours of data on 2x RTX 3090.

but so far after training for roughly 60K steps (another model trained up to 125K step - or ~10 epochs) the result is pure gibberish which clearly shows a failure of correctly aligning to the text. The speaker similarity is high though.

the modifications I have done so far boils down to using a 44.1khz config to make it compatible with my vocoder, and slightly changing how the data is being loaded : 

```python
    dataset = load_from_disk("simorq/rsr_dataset") # this is mostly the "audio | {"json": duration , text " column formatted similarly to your "lucasnewman/libritts-r-webdataset" minus the arrays where we're going to cast in the next line through the full path
    dataset = dataset.cast_column("audio", Audio(sampling_rate=44_100))
    
    print("Audio prepared, now we begin Training...")

    total_steps = 1_000_000
    batch_size = 12
    max_duration_sec = 16
    max_duration = int(max_duration_sec * SAMPLES_PER_SECOND)
    max_batch_frames = int(batch_size * max_duration)

    trainer.train(
        dataset,
        total_steps,
        batch_size=batch_size,
        max_batch_frames=max_batch_frames,
        max_duration=max_duration,
        num_workers=16,
        save_step=2500,
        restore_step=None,
    )


train()

```
i highly doubt these had anything to do with it. I tried both with the vocab Tokenization and the UTF8 method. I also tried with a bunch of different datasets, datasets that I've successfully trained solid models so they're battle-proven; nonetheless they're all producing the same nonsensical outputs.

I'd love to use this as the backbone of my future works from here on, so I really appreciate it if you tell me what could be wrong. 
Thanks!

**Comment by lucasnewman** [comment]
> i highly doubt these had anything to do with it. I tried both with the vocab Tokenization and the UTF8 method. I also tried with a bunch of different datasets, datasets that I've successfully trained solid models so they're battle-proven; nonetheless they're all producing the same nonsensical outputs.
> 
> Usually in models where explicit alignment happens (with MAS etc.) 30K steps would be enough to get basic alignment, but i wonder if that's not the case here and I am severely under training it?

Yes, 125k steps isn't enough to establish alignment -- you'll likely need 300k steps or more to start getting coherent speech. I'd recommend at least a million steps for a fully trained model.

**Comment by Respaired** [comment]
> > i highly doubt these had anything to do with it. I tried both with the vocab Tokenization and the UTF8 method. I also tried with a bunch of different datasets, datasets that I've successfully trained solid models so they're battle-proven; nonetheless they're all producing the same nonsensical outputs.
> > Usually in models where explicit alignment happens (with MAS etc.) 30K steps would be enough to get basic alignment, but i wonder if that's not the case here and I am severely under training it?
> 
> Yes, 125k steps isn't enough to establish alignment -- you'll likely need 300k steps or more to start getting coherent speech. I'd recommend at least a million steps for a fully trained model.

oof.
Yes, I found it the hard way.
is there anyway to speed up the alignment learning without relying on Forced aligners?

I'm thinking about what DiTTO is doing by leveraging the latents produced by a pretrained LLM encoder like T5 and adding a sentence level duration predictor. or maybe using a pretrained aligner with CTC as an auxiliary loss like DMO-Speech. 

otherwise training these DiT models will take forever without a cluster of H100s.

**Comment by lucasnewman** [comment]
You could consider using an approach like [FlexSpeech](https://arxiv.org/abs/2505.05159) which does an MFA step and then trains an autoregressive duration predictor for the phonemes, which would give much faster alignment.

There's not really a silver bullet to make the learned alignment train faster, but you could also try curriculum learning by training on the shortest samples first and gradually extending the duration.
